backend/src/main/java/org/youcancook/gobong/domain/user/entity/OAuthProvider.java

```diff
@@ -1,5 +1,15 @@
 package org.youcancook.gobong.domain.user.entity;
 
+import org.youcancook.gobong.domain.user.exception.OAuthProviderNotFoundException;
+
 public enum OAuthProvider {
-    KAKAO, GOOGLE
+    KAKAO, GOOGLE;
+
+    public static OAuthProvider convertToEnum(String provider) {
+        try {
+            return OAuthProvider.valueOf(provider.toUpperCase());
+        } catch (IllegalArgumentException e) {
+            throw new OAuthProviderNotFoundException();
+        }
+    }
```

Comment:
Java8 Stream을 사용해 보는 건 어떤가요? Try-catch 없이 작성할 수 있습니다
```suggestion
    public static OAuthProvider from(String provider){
        return Arrays.stream(values())
                .filter(value -> value.toString().equalsIgnoreCase(provider))
                .findFirst()
                .orElseThrow(OAuthProviderNotFoundException::new);
    }
```
나아가서 `from`이라는 메서드 이름을 사용해도 괜찮을 것 같아요.

Reply:
반영했습니다.
그런데 `try-catch`를 사용하지 않고, `Array stream`으로 선형 탐색하는 특별한 이유가 있나요?

Reply:
저는 단순하게 Nest 단계를 하나 줄여서 코드의 가독성을 올릴 수 있다고 생각해요.
그 외에도 Stream을 활용함에 있어서 확장성 등도 생각할 수 있을 것 같네요

지금은 Provider의 개수가 많지 않아서 단순 선형탐색의 시간이 크진 않겠지만, 개수가 많아진다면 `Hashmap`을 고려할 수도 있습니다.

[reference](https://velog.io/@ljinsk3/Enum%EC%9D%98-%EC%9A%94%EC%86%8C%EB%A5%BC-%EC%A1%B0%ED%9A%8C%ED%95%98%EB%8A%94-%EB%B0%A9%EB%B2%95-%EB%B9%84%EA%B5%90-%ED%8C%A9%ED%86%A0%EB%A6%AC-%EB%A9%94%EC%84%9C%EB%93%9C-vs-HashMap)